packages/ui/src/primitives/connection.tsx

```diff
@@ -3,20 +3,37 @@ import * as React from 'react';
 
 import { Spinner } from './spinner';
 
-export const Root = React.forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(function Root(
-  { children, className, ...props },
-  forwardedRef,
-) {
+function getColumnCount<T>(items: T[], maxCols: number = 6): number {
+  const numRows = Math.ceil(items.length / maxCols);
+  const itemsPerRow = Math.ceil(items.length / numRows);
+  const rows: T[][] = Array.from({ length: numRows }, () => []);
+  let currentArrayIndex = 0;
+  for (const item of items) {
+    rows[currentArrayIndex].push(item);
+    if (rows[currentArrayIndex].length === itemsPerRow) {
+      currentArrayIndex++;
+    }
+  }
```

Comment:
Likely can be simplified and just take in a count and give back a column value to be used.

Reply:
Updated https://github.com/clerk/javascript/pull/3540/commits/f4d1e9c4c012555c595518d8cd8f3e109371b921

Reply:
@alexcarpenter any chance we could use this internally for `dashboard/new` as a follow-up task? Our current implementation is pretty rough (myself and Austin just hard-coded the conditions!)

Reply:
For sure, no reason you wouldn't be able to use.